core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlUtils.java

```diff
@@ -33,32 +34,18 @@
  */
 public class RelToSqlUtils {
 
-  /** Returns list of all RexInputRef objects from the given condition. */
-  private List<RexNode> getOperandsOfTypeRexInputRefFromRexNode(RexNode conditionRex,
-      List<RexNode> inputRefRexList) {
-    if (conditionRex instanceof RexInputRef) {
-      inputRefRexList.add(conditionRex);
-    } else if (conditionRex instanceof RexCall) {
-      for (RexNode operand : ((RexCall) conditionRex).getOperands()) {
-        if (operand instanceof RexLiteral) {
-          continue;
-        } else {
-          getOperandsOfTypeRexInputRefFromRexNode(operand, inputRefRexList);
-        }
-      }
-    }
-    return inputRefRexList;
-  }
-
-  /** Returns whether an operand is Analytical Function by traversing till next project rel
+  /**
+   * Returns whether an operand is Analytical Function by traversing till next project rel
    * For ex, FilterRel e1 -> FilterRel e2 -> ProjectRel p -> TableScan ts
    * Here, we are traversing till ProjectRel p to check whether an operand of FilterRel e1
-   * is Analytical function or not. */
-  private boolean isOperandAnalyticalInFollowingProject(RelNode rel, Integer rexOperandIndex) {
+   * is Analytical function or not.
+   */
+  private static boolean isOperandAnalyticalInFollowingProject(RelNode rel,
+      Integer rexOperandIndex) {
     if (rel instanceof Project) {
       return (((Project) rel).getProjects().size() - 1) >= rexOperandIndex
           && (((Project) rel).getProjects().get(rexOperandIndex) instanceof RexOver
```

Comment:
this check is redundant, as this condition check is already handled in isAnalyticalRex. can you remove this.

Reply:
@Katakam-Phani-Mahidhar had removed that condition.Please check